.github/workflows/ci.yml

```diff
@@ -118,3 +118,19 @@ jobs:
           PGP_SECRET: ${{ secrets.PGP_SECRET }}
           SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
           SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+      
+      - name: Compile Documentation Markdown Files
+        if: startsWith(github.ref, 'refs/tags/v') # do not run for snapshots
+        run: sbt "docs/mdoc"
+
+      - name: Build Microsite
+        if: startsWith(github.ref, 'refs/tags/v') # do not run for snapshots
```

Comment:
We could actually build it always, might be a way to find some issues before a release

Reply:
Thing is, the version number provided in the docs is derived from git tags. We'll need to add some logic to derive it some other way if we are to release it on any merge to the main branch

Reply:
I suppose we could build it without actually pushing it/releasing it

Reply:
yep, that's what I meant. As for special logic, yeah I think that'd be necessary at some point (in case of documentation-only changes, for example)

Reply:
Let's do it in a next PR, I don't want to have to think about what to cache between build and release in this PR